vignettes/atlascreation.Rmd

```diff
@@ -0,0 +1,143 @@
+
+The `check_raw_counts()` function takes in a valid gene expression counts matrix and a constant max_log_value of 50. It works by checking all of the values in the gene expression matrix for integers indicating that it is raw counts matrix. However, if doubles are found, a new series of checks begin. If a counts value appears to be greater than the max_log_value, the matrix may be considered scalar normalized and if there is no value greater than the max_log_value and the counts matrix contains doubles, it is considered to be log normalized. 
+
+For this vignette, two average gene expression reference matrices will be generated and concatenated into a small reference matrix. The studies used will be GSE113049 (Single Cell RNA Sequencing Identifies TGFβ as a Key Regenerative Cue following LPS-induced Lung Injury) and GSE124952 (Cell type-specific transcriptional programs in mouse prefrontal cortex during adolescence and addiction). 
+
+```{r Reference Matrix Generation GSE113049}
+library(dplyr)
+library(Seurat)
+library(patchwork)
+library(clustifyr)
+library(tidyverse)
+library(digest)
+
+mat_Lung <- read_tsv("ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE113nnn/GSE113049/suppl/GSE113049_count_matrix.tsv.gz", skip = 1, col_names = F)
+ids_mat_Lung <- read_tsv("ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE113nnn/GSE113049/suppl/GSE113049_count_matrix.tsv.gz", n_max = 1, col_names = F)
+mat_Lung <- mat_Lung %>%
+  column_to_rownames('X1')
+mat_Lung[1:5, 1:5]
+colnames(mat_Lung) <- ids_mat_Lung[1,] %>% unlist() %>% as.vector()
+
+meta_Lung <- read_tsv("ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE113nnn/GSE113049/suppl/GSE113049_cell_metadata.tsv.gz")
+meta_Lung
+sum(colnames(mat_Lung) %in% meta_Lung$cell_type)
+ncol(mat_Lung)
+
+source("~/Reference-Matrix-Generation/R/utils/check.r")
+checkRawCounts(as.matrix(mat_Lung))
+
+GSE113049Normalized <- NormalizeData(mat_Lung)
+GSE113049Normalized
+
+new_ref_matrix <- average_clusters(mat = GSE113049Normalized, metadata = meta_Lung$cell_type, if_log = FALSE, output_log = FALSE)
+saveRDS(new_ref_matrix, "GSE113049.rds")
+```
+
+```{r Reference Matrix Generation GSE124952}
+#An example of reference matrix generation for GEO record GSE124952
+library(dplyr)
+library(Seurat)
+library(patchwork)
+library(clustifyr)
+library(tidyverse)
+library(digest)
+library(here)
```

Comment:
try not to use `here` either

Reply:
@raysinensis If we don't use the here library, would we have to include the entire function in the rmd file then?

Reply:
don't need here anymore